UIAM for Cloud org members and API keys (#4620)
elastic/docs-content: deploy-manage/api-keys/elastic-cloud-api-keys.md

# {{ecloud}} API keys [ec-api-authentication]

{{ecloud}} API keys allow you to programmatically access the following resources:

* [{{ecloud}}]({{cloud-apis}}) APIs
* [{{ecloud}} {{serverless-short}}]({{cloud-serverless-apis}}) APIs
* {applies_to}`serverless: ga` Optionally, [{{es}} {{serverless-full}}]({{es-serverless-apis}}) and [{{kib}} {{serverless-full}}]({{kib-serverless-apis}})  APIs

Only **Organization owners** can create and manage API keys. An API key is not tied to the user who created it. When creating a key, you assign it specific roles to control its access to organizational resources, including hosted deployments and serverless projects. If a user leaves the organization, the API keys they have created will still function until they expire.

You can have multiple API keys for different purposes, and you can revoke them when you no longer need them. Each organization can have up to 500 active API keys.

:::{admonition} {{es}} and {{kib}} API access 
:applies_to: ech:

By default, {{ecloud}} API keys provide access to the APIs for managing your organization, deployments, and projects. 

> In the case of {{ech}} deployments, {{ecloud}} API keys do not provide access to {{es}} or {{kib}} APIs. [Learn how to create an {{es}} API key for ECH deployments](elasticsearch-api-keys.md).

Author: ```suggestion
In {{ech}} deployments, {{ecloud}} API keys do not provide access to {{es}} or {{kib}} APIs. [Learn how to create an {{es}} API key for ECH deployments](elasticsearch-api-keys.md).
```

When an API key is nearing expiration, Elastic sends an email to the creator of the API key and each of the operational contacts. When you use an API key to authenticate, the API response header `X-Elastic-Api-Key-Expiration` indicates the key’s expiration date. You can log this value to detect API keys that are nearing expiration.

When an API key expires, it is automatically removed from the **API keys** tab.

## Applying roles to API keys [roles]

Roles grant an API key specific privileges to your {{ecloud}} organization and resources.

You can grant a cloud API key [the same types of roles that you assign to users](/deploy-manage/users-roles/cloud-organization/user-roles.md#types-of-roles): organization-level roles, cloud resource access roles, and connected cluster roles.

### Granting {{es}} and {{kib}} API access [project-access]
```{applies_to}
serverless: ga
```

When you grant **Organization owner** access, or **Cloud resource** access for one or more {{serverless-short}} projects, you can select your level of API access:

* **Cloud API** (default): Grants access to only [{{ecloud}} {{serverless-short}}]({{cloud-serverless-apis}}) APIs.

> * **Cloud, {{es}} and {{kib}} API**: Grants access to the relevant projects and configuration options through the [{{ecloud}} {{serverless-short}}]({{cloud-serverless-apis}}), and [{{es}} {{serverless-short}}]({{es-serverless-apis}}), and [{{kib}} {{serverless-short}}]({{kib-serverless-apis}}) API endpoints for the relevant projects. 

Reviewer: ```suggestion
* **Cloud, {{es}} and {{kib}} API**: Grants access to the relevant projects and configuration options through the [{{ecloud}} {{serverless-short}}]({{cloud-serverless-apis}}), [{{es}} {{serverless-short}}]({{es-serverless-apis}}), and [{{kib}} {{serverless-short}}]({{kib-serverless-apis}}) API endpoints for the relevant projects. 
```

When an API key expires, it is automatically removed from the **API keys** tab.

## Applying roles to API keys [roles]

Roles grant an API key specific privileges to your {{ecloud}} organization and resources.

You can grant a cloud API key [the same types of roles that you assign to users](/deploy-manage/users-roles/cloud-organization/user-roles.md#types-of-roles): organization-level roles, cloud resource access roles, and connected cluster roles.

### Granting {{es}} and {{kib}} API access [project-access]
```{applies_to}
serverless: ga
```

When you grant **Organization owner** access, or **Cloud resource** access for one or more {{serverless-short}} projects, you can select your level of API access:

| Access | Grant |
| --- | --- |
| **Cloud API** (default) | Grants access to only [{{ecloud}}]({{cloud-apis}}) and [{{ecloud}} {{serverless-short}}]({{cloud-serverless-apis}}) APIs. No direct access to project {{es}} or {{kib}} API endpoints. |

> | **Cloud, {{es}} and {{kib}} API** | Grants the following access:<br><br>•[{{ecloud}}]({{cloud-apis}}) and [{{ecloud}} {{serverless-short}}]({{cloud-serverless-apis}}) APIs.<br><br>• [{{es}} {{serverless-short}}]({{es-serverless-apis}}), and [{{kib}} {{serverless-short}}]({{kib-serverless-apis}}) API endpoints for the relevant projects. |

Reviewer 2: nit: i wouldn't use terminal punctuation in the list items, but very nitty indeed

Using {{ecloud}} keys for project-level API access, rather than [granting keys from within each {{serverless-short}} project](serverless-project-api-keys.md), allows you to create keys that can interact with multiple projects, and manage API access centrally from the {{ecloud}} console.

> When granting Cloud resource access, you can apply a [predefined role](/deploy-manage/users-roles/cloud-organization/user-roles.md#general-assign-user-roles-table) or [custom role](/deploy-manage/users-roles/serverless-custom-roles.md) to granularly control access to the specified resources. The role that you select controls the resources that you can access in all relevant APIs. 

Reviewer 2: ```suggestion
When granting Cloud resource access, you can apply a [predefined role](/deploy-manage/users-roles/cloud-organization/user-roles.md#general-assign-user-roles-table) or [custom role](/deploy-manage/users-roles/serverless-custom-roles.md) to granularly control access to the specified resources. The selected role controls access to resources in all relevant APIs. 
```

Reviewer 2: made this suggestion partly because the two "yous" were slightly confusing

#### Considerations

> All roles include permissions to both Cloud APIs and {{es}} and {{kib}} APIs. Your **API access** selection limits the role's scope, and, in some cases, can disable core functionality.

Reviewer 2: ```suggestion
All roles include permissions for Cloud APIs, as well as {{es}} and {{kib}} APIs. Your **API access** selection limits the role's scope, and, in some cases, can disable core functionality.
```
